internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/issuecreator/JIRAIssueCreator.java

```diff
@@ -0,0 +1,246 @@
+/*
+ *
+ *   Copyright (c) 2018, WSO2 Inc., WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package org.wso2.security.tools.scanner.dependency.js.issuecreator;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.security.tools.scanner.dependency.js.constants.IssueCreatorConstants;
+import org.wso2.security.tools.scanner.dependency.js.constants.JSScannerConstants;
+import org.wso2.security.tools.scanner.dependency.js.exception.ConfigParserException;
+import org.wso2.security.tools.scanner.dependency.js.exception.IssueCreatorException;
+import org.wso2.security.tools.scanner.dependency.js.utils.ConfigParser;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.naming.AuthenticationException;
+
+/**
+ * JIRA Issue Creator. This class responsible for essential functions for creating a JIRA issue ticket and
+ * update existing tickets with newly scan reports.
+ */
+public class JIRAIssueCreator extends IssueCreator {
```

Comment:
Throughout the code, you seem to have interchangeably used "issue" and "ticket" to refer JIRA tickets. Shall we stick with one? Otherwise it is bit confusing. Most appropriate would be 'ticket'